modules/evm/src/lib.rs

```diff
@@ -1494,3 +1521,94 @@ impl<T: Config + Send + Sync> SignedExtension for SetEvmOrigin<T> {
 		Ok(())
 	}
 }
+
+#[derive(Clone, RuntimeDebug, PartialEq, Encode, Decode, TypeInfo)]
+pub enum EvmTask<T: Config> {
+	// TODO: update
+	Schedule {
+		from: EvmAddress,
+		target: EvmAddress,
+		input: Vec<u8>,
+		value: BalanceOf<T>,
+		gas_limit: u64,
+		storage_limit: u32,
+	},
+	Remove {
+		caller: EvmAddress,
+		contract: EvmAddress,
+		maintainer: EvmAddress,
+	},
+}
+
+impl<T: Config> DispatchableTask for EvmTask<T> {
+	fn dispatch(self, weight: Weight) -> TaskResult {
+		match self {
+			// TODO: update
+			EvmTask::Schedule { .. } => {
+				// check weight and call `scheduled_call`
+				TaskResult {
+					result: Ok(()),
+					used_weight: 0,
+					finished: false,
+				}
+			}
+			EvmTask::Remove {
+				caller,
+				contract,
+				maintainer,
+			} => {
+				// default limit 100
+				let limit = cmp::min(
+					weight
+						.checked_div(<T as frame_system::Config>::DbWeight::get().write)
+						.unwrap_or(100),
+					100,
+				) as u32;
+
+				match <AccountStorages<T>>::remove_prefix(contract, Some(limit)) {
+					AllRemoved(count) => {
+						let res = Pallet::<T>::refund_storage(&caller, &contract, &maintainer);
+						log::debug!(
+							target: "evm",
+							"EvmTask::Remove: [from: {:?}, contract: {:?}, maintainer: {:?}, count: {:?}, result: {:?}]",
+							caller, contract, maintainer, count, res
+						);
+
+						// Remove account after all of the storages are cleared.
+						Accounts::<T>::take(contract);
+
+						TaskResult {
+							result: res,
+							used_weight: <T as frame_system::Config>::DbWeight::get()
```

Comment:
we should do some real benchmarking here to include the weights of refund and remove accounts

this can happen later